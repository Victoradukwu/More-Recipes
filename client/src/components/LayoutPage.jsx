import React from 'react';
import { Route, Switch } from 'react-router-dom';

import HomePage from './recipes/HomePage.jsx';
import connectedManageRecipePage from './recipes/ManageRecipePage.jsx';
import SignUpPage from './users/SignupPage.jsx';
import SignInPage from './users/SigninPage.jsx';
import UserProfile from './users/UserProfile.jsx';
import SignOutPage from './users/SignOutPage.jsx';
import NotFoundPage from './NotFoundPage.jsx';
import Navbar from './Navbar';
import RecipeDetailsPage from './recipes/recipeDetails/RecipeDetailsPage.jsx';
import RecipeSearchPage from './recipes/RecipeSearchPage';
import UserRecipesPage from './recipes/UserRecipesPage';
import UserFavoritesPage from './recipes/UserFavoritesPage';


const LayoutPage = () => (
  <div className="body">
    <Navbar />
    <main className="main" style={{ paddingTop: '80px' }}>
      <Switch>
        <Route path="/" exact component={HomePage} />
        <Route path="/favorites" component={UserFavoritesPage} />
        <Route exact path="/signup" component={SignUpPage} />
        <Route path="/signin" component={SignInPage} />
        <Route path="/signout" component={SignOutPage} />
        <Route path="/user" component={UserProfile} />
        <Route path="/recipe/:id" component={connectedManageRecipePage} />
        <Route path="/recipe" exact component={connectedManageRecipePage} />
        <Route path="/myRecipes" component={UserRecipesPage} />
        <Route path="/recipeDetail/:id" component={RecipeDetailsPage} />
        <Route path="/recipeSearch" component={RecipeSearchPage} />
        <Route path="*" component={NotFoundPage} />
      </Switch>
    </main>
    <footer className="footer text-center">
      <p>&copy; More-Recipes {(new Date()).getFullYear()}</p>
      <div>
        <span className="fa fa-envelope">: more_recipes@andela.com</span>
        &nbsp; &nbsp;
        <span className="fa fa-phone">: +2348012345678</span>
      </div>
    </footer>
  </div>
);


export default LayoutPage;
