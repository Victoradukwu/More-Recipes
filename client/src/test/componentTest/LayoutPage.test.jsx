import React from 'react';
import { shallow } from 'enzyme';
import LayoutPage from '../../components/LayoutPage';


describe('LayoutPage', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<LayoutPage />);
    expect(wrapper.find('main').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
