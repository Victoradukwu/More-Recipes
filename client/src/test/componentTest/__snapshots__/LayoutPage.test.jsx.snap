// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutPage matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LayoutPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <main
          className="main"
          style={
            Object {
              "paddingTop": "80px",
            }
          }
        >
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/favorites"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />
            <Route
              component={[Function]}
              path="/signin"
            />
            <Route
              component={[Function]}
              path="/signout"
            />
            <Route
              component={[Function]}
              path="/user"
            />
            <Route
              component={[Function]}
              path="/recipe/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/recipe"
            />
            <Route
              component={[Function]}
              path="/myRecipes"
            />
            <Route
              component={[Function]}
              path="/recipeDetail/:id"
            />
            <Route
              component={[Function]}
              path="/recipeSearch"
            />
            <Route
              component={[Function]}
              path="*"
            />
          </Switch>
        </main>,
        <footer
          className="footer text-center"
        >
          <p>
            © More-Recipes 2018
          </p>
          <div>
            <span
              className="fa fa-envelope"
            >
              : more_recipes@andela.com
            </span>
               
            <span
              className="fa fa-phone"
            >
              : +2348012345678
            </span>
          </div>
        </footer>,
      ],
      "className": "body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/favorites"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />
            <Route
              component={[Function]}
              path="/signin"
            />
            <Route
              component={[Function]}
              path="/signout"
            />
            <Route
              component={[Function]}
              path="/user"
            />
            <Route
              component={[Function]}
              path="/recipe/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/recipe"
            />
            <Route
              component={[Function]}
              path="/myRecipes"
            />
            <Route
              component={[Function]}
              path="/recipeDetail/:id"
            />
            <Route
              component={[Function]}
              path="/recipeSearch"
            />
            <Route
              component={[Function]}
              path="*"
            />
          </Switch>,
          "className": "main",
          "style": Object {
            "paddingTop": "80px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/favorites"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />,
              <Route
                component={[Function]}
                path="/signin"
              />,
              <Route
                component={[Function]}
                path="/signout"
              />,
              <Route
                component={[Function]}
                path="/user"
              />,
              <Route
                component={[Function]}
                path="/recipe/:id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/recipe"
              />,
              <Route
                component={[Function]}
                path="/myRecipes"
              />,
              <Route
                component={[Function]}
                path="/recipeDetail/:id"
              />,
              <Route
                component={[Function]}
                path="/recipeSearch"
              />,
              <Route
                component={[Function]}
                path="*"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/signin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/signout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/user",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/recipe/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/recipe",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/myRecipes",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/recipeDetail/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/recipeSearch",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "*",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              © More-Recipes 2018
            </p>,
            <div>
              <span
                className="fa fa-envelope"
              >
                : more_recipes@andela.com
              </span>
                 
              <span
                className="fa fa-phone"
              >
                : +2348012345678
              </span>
            </div>,
          ],
          "className": "footer text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "© More-Recipes 2018",
            },
            "ref": null,
            "rendered": "© More-Recipes 2018",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="fa fa-envelope"
                >
                  : more_recipes@andela.com
                </span>,
                "   ",
                <span
                  className="fa fa-phone"
                >
                  : +2348012345678
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": ": more_recipes@andela.com",
                  "className": "fa fa-envelope",
                },
                "ref": null,
                "rendered": ": more_recipes@andela.com",
                "type": "span",
              },
              "   ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": ": +2348012345678",
                  "className": "fa fa-phone",
                },
                "ref": null,
                "rendered": ": +2348012345678",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <main
            className="main"
            style={
              Object {
                "paddingTop": "80px",
              }
            }
          >
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/favorites"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                path="/signin"
              />
              <Route
                component={[Function]}
                path="/signout"
              />
              <Route
                component={[Function]}
                path="/user"
              />
              <Route
                component={[Function]}
                path="/recipe/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/recipe"
              />
              <Route
                component={[Function]}
                path="/myRecipes"
              />
              <Route
                component={[Function]}
                path="/recipeDetail/:id"
              />
              <Route
                component={[Function]}
                path="/recipeSearch"
              />
              <Route
                component={[Function]}
                path="*"
              />
            </Switch>
          </main>,
          <footer
            className="footer text-center"
          >
            <p>
              © More-Recipes 2018
            </p>
            <div>
              <span
                className="fa fa-envelope"
              >
                : more_recipes@andela.com
              </span>
                 
              <span
                className="fa fa-phone"
              >
                : +2348012345678
              </span>
            </div>
          </footer>,
        ],
        "className": "body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/favorites"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                path="/signin"
              />
              <Route
                component={[Function]}
                path="/signout"
              />
              <Route
                component={[Function]}
                path="/user"
              />
              <Route
                component={[Function]}
                path="/recipe/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/recipe"
              />
              <Route
                component={[Function]}
                path="/myRecipes"
              />
              <Route
                component={[Function]}
                path="/recipeDetail/:id"
              />
              <Route
                component={[Function]}
                path="/recipeSearch"
              />
              <Route
                component={[Function]}
                path="*"
              />
            </Switch>,
            "className": "main",
            "style": Object {
              "paddingTop": "80px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  path="/favorites"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/signup"
                />,
                <Route
                  component={[Function]}
                  path="/signin"
                />,
                <Route
                  component={[Function]}
                  path="/signout"
                />,
                <Route
                  component={[Function]}
                  path="/user"
                />,
                <Route
                  component={[Function]}
                  path="/recipe/:id"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/recipe"
                />,
                <Route
                  component={[Function]}
                  path="/myRecipes"
                />,
                <Route
                  component={[Function]}
                  path="/recipeDetail/:id"
                />,
                <Route
                  component={[Function]}
                  path="/recipeSearch"
                />,
                <Route
                  component={[Function]}
                  path="*"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/signin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/signout",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/user",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/recipe/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/recipe",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/myRecipes",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/recipeDetail/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/recipeSearch",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "*",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                © More-Recipes 2018
              </p>,
              <div>
                <span
                  className="fa fa-envelope"
                >
                  : more_recipes@andela.com
                </span>
                   
                <span
                  className="fa fa-phone"
                >
                  : +2348012345678
                </span>
              </div>,
            ],
            "className": "footer text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "© More-Recipes 2018",
              },
              "ref": null,
              "rendered": "© More-Recipes 2018",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="fa fa-envelope"
                  >
                    : more_recipes@andela.com
                  </span>,
                  "   ",
                  <span
                    className="fa fa-phone"
                  >
                    : +2348012345678
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": ": more_recipes@andela.com",
                    "className": "fa fa-envelope",
                  },
                  "ref": null,
                  "rendered": ": more_recipes@andela.com",
                  "type": "span",
                },
                "   ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": ": +2348012345678",
                    "className": "fa fa-phone",
                  },
                  "ref": null,
                  "rendered": ": +2348012345678",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
